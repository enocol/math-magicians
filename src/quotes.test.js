import React from 'react';
import { render } from '@testing-library/react';
import Quotes from './components/quotes';

describe('Quotes component', () => {
  it('renders correctly when loading', () => {
    const { asFragment } = render(<Quotes state="loading" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly when an error occurs', () => {
    const { asFragment } = render(<Quotes state="error" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with quotes', () => {
    const { asFragment } = render(<Quotes state="success" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
