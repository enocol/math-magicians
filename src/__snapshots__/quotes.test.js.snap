// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component renders correctly when an error occurs 1`] = `
<DocumentFragment>
  <div
    class="status"
  >
    Loading......
  </div>
</DocumentFragment>
`;

exports[`Quotes component renders correctly when loading 1`] = `
<DocumentFragment>
  <div
    class="status"
  >
    Loading......
  </div>
</DocumentFragment>
`;

exports[`Quotes component renders correctly with quotes 1`] = `
<DocumentFragment>
  <div
    class="status"
  >
    Loading......
  </div>
</DocumentFragment>
`;
